File: src/org/thoughtcrime/securesms/notifications/MessageNotifier.java
package org.thoughtcrime.securesms.notifications;

import android.content.Context;

import org.thoughtcrime.securesms.recipients.Recipient;

import androidx.annotation.NonNull;

public interface MessageNotifier {
  void setVisibleThread(long threadId);
  void setLastDesktopActivityTimestamp(long timestamp);
  void notifyMessageDeliveryFailed(Context context, Recipient recipient, long threadId);
  void cancelDelayedNotifications();
  void updateNotification(@NonNull Context context);
  void updateNotification(@NonNull Context context, long threadId);
  void updateNotification(@NonNull Context context, long threadId, boolean signal);
  void updateNotification(@android.support.annotation.NonNull Context context, boolean signal, int reminderCount);
  void clearReminder(@NonNull Context context);
}

File: libsession/src/main/java/org/session/libsession/messaging/sending_receiving/notifications/MessageNotifier.kt
package org.session.libsession.messaging.sending_receiving.notifications

import android.content.Context
import org.session.libsession.utilities.recipients.Recipient

interface MessageNotifier {
    fun setVisibleThread(threadId: Long)
    fun setLastDesktopActivityTimestamp(timestamp: Long)
    fun notifyMessageDeliveryFailed(context: Context?, recipient: Recipient?, threadId: Long)
    fun cancelDelayedNotifications()
    fun updateNotification(context: Context)
    fun updateNotification(context: Context, threadId: Long)
    fun updateNotification(context: Context, threadId: Long, signal: Boolean)
    fun updateNotification(context: Context, signal: Boolean, reminderCount: Int)
    fun clearReminder(context: Context)
}